From: Roland Hieber <rohieb@rohieb.name>
Date: Wed, 6 Dec 2017 13:48:50 +0100
Subject: [PATCH] ARM: dts: bcm2835-rpi: re-enable booting from SD card

Kernel commit a19adf8d86e822eba502486524127595372c85f6 [0] changed the
RPi1 device tree from &sdhci to &sdhost, which was imported in barebox
commit d14b844b08635c717fb52a294ed8d6872e260315. It seems that barebox
does not have a driver for brcm,bcm2835-sdhost (yet), which resulted in
barebox unable to boot from SD card. However, the driver for
brcm,bcm2835-sdhci worked fine in the past.

Although the upstream change was made for good reasons, the simplest
change for barebox for now is to revert to using &sdhci and disable
&sdhost.

Thanks to Alexander Dahl for raising this issue on the DistroKit
mailing list!

[0]: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=a19adf8d86e822eba502486524127595372c85f6

Forwarded: barebox@lists.infradead.org; <20171206151233.29342-1-r.hieber@pengutronix.de>
Signed-off-by: Roland Hieber <rohieb@rohieb.name>
---
 arch/arm/dts/bcm2835-rpi.dts | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/dts/bcm2835-rpi.dts b/arch/arm/dts/bcm2835-rpi.dts
index 0095f58a3ce8..ab2d6dff64f5 100644
--- a/arch/arm/dts/bcm2835-rpi.dts
+++ b/arch/arm/dts/bcm2835-rpi.dts
@@ -9,3 +9,10 @@
 		reg = <0x0 0x0>;
 	};
 };
+&sdhci {
+	status = "okay";
+};
+
+&sdhost {
+	status = "disabled";
+};
